File: backend/assetflow-company-service/src/main/java/com/assetflow/company/response/ApiResponse.java
package com.assetflow.company.response;

import java.time.Instant;

public record ApiResponse<T>(boolean success, T data, String message, Instant timestamp) {
	public static <T> ApiResponse<T> ok(T data) {
		return new ApiResponse<>(true, data, null, Instant.now());
	}
}

File: dotnet-backend/Assetflow.CompanyService/Models/ApiResponse.cs
using System.Text.Json.Serialization;

namespace Assetflow.CompanyService.Models
{
    public class ApiResponse<T>
    {
        [JsonPropertyName("success")]
        public bool Success { get; set; }

        [JsonPropertyName("data")]
        public T? Data { get; set; }

        [JsonPropertyName("message")]
        public string? Message { get; set; }

        [JsonPropertyName("timestamp")]
        public DateTime Timestamp { get; set; }

        public static ApiResponse<T> Ok(T data)
        {
            return new ApiResponse<T>
            {
                Success = true,
                Data = data,
                Message = null,
                Timestamp = DateTime.UtcNow
            };
        }
    }
}
